client/src/components/FlashCard/CreateFlashcard.js

```diff
@@ -0,0 +1,109 @@
+import React, {useState} from "react"
+import { Button, Stack, TextField } from "@mui/material"
+import axios from 'axios'
+
+const CreateFlashcard = ({ userId, deckId }) => {
+  console.log(`[CreateFlashcard] deckId is ${deckId}`)
+  const [formValue, setFormValue] = useState({})
+  const [errors, setErrors] = useState({
+    'frontText': 'Front text input error',
+    'frontImage': 'Front image url error',
+    'backText': 'Back text input error',
+    'backImage': 'Back image url error'
+  })
+
+  const isUrl = (value) => {
+    const re = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)/
+    return re.test(value)
+  }
+
+  function validateProperty(fieldName, fieldValue) {
+    const fieldValueTrimmed = fieldValue.trim()
+    if (fieldValueTrimmed === '') {
+      return 'Field cannot be empty'
+    } else if(fieldName === "frontImage" || fieldName === "backImage") {
+      const validateUrl = fieldName.isUrl()
+      if (validateUrl === false) {
+        return 'Invalid Url entered'
+      }
+    }
+    return null
+  }
+
+  const handleChange = (event) => {
+    event.preventDefault()
+    console.log("[CreateFlashcard] onChange ", event)
+    if (validateProperty(event.target.name, event.target.value)) {
+    const currentValues = formValue
+    currentValues[event.target.name] = event.target.value
+    setFormValue(currentValues)
+  } else {
+    setErrors(formValue)
+    return ' event handling error, one or more card fields contains invalid information'
+  }
+}
+  const handleSubmit = async (event) => {
+    console.log("[CreateFlashcard] onSubmit ", event)
+    event.preventDefault()
+    try {
+      const response = await axios.post(`http://localhost:8000/decks/${deckId}/cards`, formValue, { headers: { user: userId } })
+      console.log(`[createflashcard] response submit ${response.status}`)
+    } catch (err) {
+      console.log(`response error ${err.status}`)
+    } if (err.status) {
```

Comment:
This line throws an error that prevents the app from running. It's also deeply confusing to have an if statement start on the last line of a different scope.

Reply:
I was attempting to let user know the deck was not found as part of the error. I deleted this line now. Also, i do understand why the if statement is totally misplaced there.